aws/resource_aws_dynamodb_table.go

```diff
@@ -360,10 +373,28 @@ func resourceAwsDynamoDbTableCreate(d *schema.ResourceData, meta interface{}) er
 
 func resourceAwsDynamoDbTableUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).dynamodbconn
-
+	if d.HasChange("billing_mode") && !d.IsNewResource() {
+		req := &dynamodb.UpdateTableInput{
+			TableName:   aws.String(d.Id()),
+			BillingMode: aws.String(d.Get("billing_mode").(string)),
+		}
+		if d.Get("billing_mode").(string) == dynamodb.BillingModeProvisioned {
+			req.ProvisionedThroughput = expandDynamoDbProvisionedThroughput(map[string]interface{}{
+				"read_capacity":  d.Get("read_capacity"),
+				"write_capacity": d.Get("write_capacity"),
+			})
+		}
+		_, err := conn.UpdateTable(req)
```

Comment:
Ideally the Update function would ever call UpdateTable once, however I see the rest of the function logic follows the duplicate call pattern across attribute changes, so this is okay for now.